frontend/public/components/console-app.jsx

```diff
@@ -0,0 +1,116 @@
+import CloudShell from '@console/app/src/components/cloud-shell/CloudShell';
+import store from '../redux';
+import AppContents from './app-contents';
+import useEventListener from './utils/useEventListener';
+import { Helmet } from 'react-helmet';
+import { Page } from '@patternfly/react-core';
+import { ConsoleNotifier } from './console-notifier';
+import { ConnectedNotificationDrawer } from './notification-drawer';
+import { getBrandingDetails, Masthead } from './masthead';
+import { Navigation } from './nav';
+
+export const ConsoleApp = ({ location, match }) => {
+  const breakpointMD = 768;
+  const NOTIFICATION_DRAWER_BREAKPOINT = 1800;
+
+  const isDesktop = () => window.innerWidth >= breakpointMD;
+  const isLargeLayout = () => window.innerWidth >= NOTIFICATION_DRAWER_BREAKPOINT;
+
+  const [previousNavOpen, setPreviousNavOpen] = React.useState(isDesktop());
+  const [isNavOpen, setIsNavOpen] = React.useState(isDesktop());
+
+  const [previousDrawerInline, setPreviousDrawerInline] = React.useState(isLargeLayout());
+  const [isDrawerInline, setIsDrawerInline] = React.useState(isLargeLayout());
+
+  const usePrevious = (value) => {
+    const ref = React.useRef();
+    React.useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  };
+
+  const prevLocation = usePrevious({ location });
+  const prevMatch = usePrevious({ match });
+
+  React.useEffect(() => {
+    const oldLocation = _.omit(prevLocation, ['key']);
+    const newLocation = _.omit(location, ['key']);
+    if (_.isEqual(newLocation, oldLocation) && _.isEqual(match, prevMatch)) {
+      return;
+    }
+    // two way data binding :-/
+    const { pathname } = location;
+    store.dispatch(UIActions.setCurrentLocation(pathname));
+  }, [location, match, prevLocation, prevMatch]);
+
+  const onNavToggle = () => {
+    // Some components, like svg charts, need to reflow when nav is toggled.
+    // Fire event after a short delay to allow nav animation to complete.
+    setTimeout(() => {
+      window.dispatchEvent(new Event('sidebar_toggle'));
+    }, 100);
+    setIsNavOpen(!isNavOpen);
+  };
+
+  const onNotificationDrawerToggle = () => {
+    if (isLargeLayout()) {
+      // Fire event after the drawer animation speed delay.
+      setTimeout(() => {
+        window.dispatchEvent(new Event('sidebar_toggle'));
+      }, 250);
+    }
+  };
+
+  const onNavSelect = () => {
+    //close nav on mobile nav selects
+    if (!isDesktop()) {
+      setIsNavOpen(false);
+    }
+  };
+
+  const onResize = () => {
+    const isCurrentDesktop = isDesktop();
+    const isCurrentDrawerInline = isLargeLayout();
+    if (previousNavOpen !== isCurrentDesktop) {
+      setIsNavOpen(isCurrentDesktop);
+      setPreviousNavOpen(isCurrentDesktop);
+    }
+    if (previousDrawerInline !== isCurrentDrawerInline) {
+      setIsDrawerInline(isCurrentDrawerInline);
+      setPreviousDrawerInline(isCurrentDrawerInline);
+    }
+  };
+
+  useEventListener('resize', onResize);
```

Comment:
I'd actually vouch for a simple direct `React.useEffect` call here :smiley:

Reply:
Why?